config.go

```diff
@@ -443,21 +442,15 @@ type moduleConfig struct {
 	// environKeys allow overwriting of existing values.
 	environKeys map[string]int
 
-	// preopenFD has the next FD number to use
-	preopenFD uint32
-	// preopens are keyed on file descriptor and only include the Path and FS fields.
-	preopens map[uint32]*wasm.FileEntry
-	// preopenPaths allow overwriting of existing paths.
-	preopenPaths map[string]uint32
+	*wasm.FSConfig
 }
 
 func NewModuleConfig() ModuleConfig {
 	return &moduleConfig{
 		startFunctions: []string{"_start"},
 		environKeys:    map[string]int{},
-		preopenFD:      uint32(3), // after stdin/stdout/stderr
-		preopens:       map[uint32]*wasm.FileEntry{},
-		preopenPaths:   map[string]uint32{},
+
+		FSConfig: wasm.NewFSConfig(),
```

Comment:
This also shouldn't need to be exported I think..

```suggestion
		fs: wasm.NewFSConfig(),
```

Reply:
It is not exporting. Just pass the value to the field. Anonymous embedding requires the same name as the struct name for initialization.

Reply:
we don't tend to do anonymous embedding is the main point, what I meant is when that is removed, let's not assign an uppercased/exported field name as that would be distracting as we never do that.

Reply:
fwiw ack this is bikeshed and also why we need to add rationale or some style notes.

we get some benefit of fast review and almost no accidental export accidents by always using unexported field names. we don't need to think too much if the field is the same name as anonymous or is anonymous when we have no anonymous. We also don't need to scan to see if the type is exported or not, and sometimes it is off diff anyway. Not saying all projects do this, but we are fairly consistent and do (should) comment when we do anything to the contrary. Once in the groove it is great because you can scan through a lot of changes fast without thinking so much all the time. OTOH projects don't always do this so it isn't a portable approach.

Anyway I thought it worth mentioning to you, as appreciate your help and new to the style here.

Reply:
Yes, I don't have a plan to export it.